sharedbuffers/mapped_struct.py

```diff
@@ -494,12 +494,19 @@ def _is_eq(self, other):
 
         return True
 
-    def __richcmp__(self, other, op):
-        if op != Py_EQ and op != Py_NE:
-            return False   # No comparison possible
+    if cython.compiled:
```

Comment:
This conditional method declaration will confuse Cython. You have to always define `__richcmp__`, do the conditional inside.

Reply:
Looking at the annotated html, there doesn't seem to be anything wrong with it.

Reply:
@gefarion experienced the issue, it's not in the annotated html because what happens is that it doesn't generate the required fields in the type descriptor. The issue is in the boilerplate code it generates to describe the type for Python. It basically won't invoke this method when doing comparisons.

You have to do it like:

```  python
class Whatever:
    def __richcmp__(self, other, op):
        ...

    if not cython.compiled:
        def _cmp(self, other):
            ...

        def _ne(self, other):
            ...

        def _eq(self, other):
            ...

if not cython.compiled:
    for orig, new in (('_cmp', '__cmp__'), ('_eq', '__eq__'), ('_ne', '__ne__')):
        setattr(proxied_dict, new, getattr(proxied_dict, orig))
```

So only what's used in pure python mode is conditional on `cython.compiled`.

Pure python will have the richcmp method, but it doesn't hurt. And you can alwas delattr if you need to.